File: Examples/cpp/VoltageModeExample/src/Robot.cpp
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

#include <iostream>
#include <string>

#include <IterativeRobot.h>
#include "Joystick.h"
#include "digilent/DMC60/WPI_DMC60C.h"

/**
 * This demo uses a joystick to control a DMC60C to drive in open loop voltage mode.
 * When the A button is held, the motor is driven based on the left joystick position.
 * When the start button is pressed, the motor is stopped.
 */

class Robot: public frc::IterativeRobot {
public:
	//Create a joystick object to control the motor.
	Joystick * _joy = new Joystick(0);

	//Create a DMC60C object with device number 1.
	DMC60::WPI_DMC60C * _dmc = new DMC60::WPI_DMC60C(1);

	bool _btn2 = false;
	bool _startbutton = false;

	void RobotInit(){
		//Disable limit switches.
		_dmc->setLimitSwitches(true, false, false);
	}

	void DisabledInit(){
		//Disable motor.
		_dmc->Disable();
	}
	void TeleopInit() {
	}

	void TeleopPeriodic() {
		bool A = _joy->GetRawButton(1);

		double stick = -1 * _joy->GetY();//Scale the joystick -1 to 1.

		if (A) {//A button on XBox controller.
			_dmc->driveVoltage(stick * 100);//Scale duty cycle from -100 to 100.
		}
		else{
			_dmc->Disable();
		}

	}



private:
};


START_ROBOT_CLASS(Robot)

File: Examples/Java/JavaVoltageMode/src/org/usfirst/frc/team1234/robot/Robot.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

package org.usfirst.frc.team1234.robot;

import com.dmc60c.DMC60C;
import edu.wpi.first.wpilibj.IterativeRobot;
import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

/**
 * This project demonstrates the use of the Open-Loop driveVoltage function.
 * It requires an XBox controller and a DMC60C. When A is pressed, the motor spins
 * depending on the left joystick position.
 */
public class Robot extends IterativeRobot {
	private Joystick _joy = new Joystick(0);//Controller on USB slot 0
	private DMC60C _dmc60c = new DMC60C(2);//DMC60C device with devicenum 2
	private double YAxis;
	private boolean A;
	/**
	 * This function is run when the robot is first started up and should be
	 * used for any initialization code.
	 */
	@Override
	public void robotInit() {
	}

	/**
	 * This function is called when teleop mode is enabled
	 */
	@Override
	public void teleopInit() {
	}

	/**
	 * This function is called periodically during operator control.
	 */
	@Override
	public void teleopPeriodic() {
		YAxis = _joy.getY()*-100;//Get left joystick Y axis. 1 to -1 scaled to -100 to 100
		A = _joy.getRawButton(1);//Get 'A' button
		if (A) {//If 'A' button is held
			_dmc60c.DriveVoltage(YAxis);
			SmartDashboard.putNumber("Duty Cycle", YAxis);
		}
		else {//No button is pressed
			_dmc60c.disable();
		}
	}
	
	public void disabledInit() {
		_dmc60c.disable();
	}
}
